### PR TITLE
Fix email & date inputs in tests

### PR DESCRIPTION
**Pull request by CarsonF**
- Patch faker email to be a more unique, valid email
- [tests] Fix complex objects being passed in with variables

Related to #1107

### CODE REVIEW

File: src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts [hunk 1]
```diff
@@ -100,8 +100,16 @@ export class SyncBudgetRecordsToFundingPartners
...
+      let mouStartYear = partnership.mouStart.value.year;
+      let mouEndYear = partnership.mouEnd.value.year;
+
+      if (!mouStartYear || !mouEndYear) {
+        mouStartYear = CalendarDate.fromISO(
+          partnership.mouStart.value.toString()
+        ).year;
+        mouEndYear = CalendarDate.fromISO(partnership.mouEnd.value.toString())
+          .year;
+      }
```
Comment (CarsonF) [inline src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts:112]: Thanks for finding this @zkhin. This shouldn't be necessary though. It looks like the `mouStart/End.value` are strings but they TS says they are CalendarDates. TS types should always match what the actual types are, if it doesn't that's a problem.
Looking into why the ISO strings are not being transformed now.

Comment (zkhin) [reply src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts:112]: Yes, that's the weird part.

Comment (CarsonF) [reply src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts:112]: The transformation gets applied by the `ValidationPipe`. That's why it was working in the other branch with that pipe re-enabled in tests.

**Comment by CarsonF** [comment]
Ok I'm closing and going to rebase this into your branch @zkhin

**Pull request closed without merging**